src/lib/core/exception.h

```diff
@@ -231,12 +248,20 @@ class CryptoError: public Exception {
 	}
 
 	virtual void raise() { throw this; }
+
+	virtual struct error *dup() const { return new CryptoError; }
 };
 
 class RaftError: public Exception {
 public:
+	RaftError()
+		:Exception(&type_RaftError, NULL, 0)
```

Comment:
Even though other functions in this file, do the same, I think that we should actually have a space after ':' like checkpatch recommends. Please fix. Also it could be written in one line:

```C++
RaftError() : Exception(&type_RaftError, NULL, 0) {}
```

Reply:
Fixed.